### ISSUE TITLE
Multiple areas of Mautic throw a 500 error

### ISSUE THREAD

**Issue by RCheesley** [mautic/mautic#9762]
## Bug Description
I came across this issue when running the Cypress DWC tests and found it to also be reported in Emails (see #9646 #9755 and /https://forum.mautic.org/t/notices-trying-to-access-array-offset-on-value-of-type-null/18008).

I found it by trying to open a DWC item that was created but others find it with using certain email templates (with the legacy builder) as listed in the forum thread.

| Q   | A
| --- | ---
| Mautic version | 3.3.1
| PHP version | 7.4
| Browser | Chrome

### Steps to reproduce
1. Create dynamic content
2. Click to edit dynamic content
3. Notice a 503 error and log entry below
 
### Log errors

`mautic.NOTICE: PHP Notice - Trying to access array offset on value of type null - in file /var/www/html/app/bundles/EmailBundle/Views/FormTheme/Email/_emailform_dynamicContent_entry_filters_entry_filters_entry_widget.html.php - at line 14 {"mauticTemplateVars":{"app":"[object] (Symfony\\Bundle\\FrameworkBundle\\Templating\\GlobalVariables: {})","value":null,"attr":{"class":"form-control"},"form":"[object] (Symfony\\Component\\Form\\FormView: null)","id":"emailform_dynamicContent_0_filters_0_filters___name__","name":"__name__","full_name":"emailform[dynamicContent][0][filters][0][filters][__name__]","disabled":false,"label":false,"label_format":null,"multipart":false,"block_prefixes":["form","dynamic_content_filter_entry_filters","_emailform_dynamicContent_entry_filters_entry_filters_entry"],"unique_block_prefix":"_emailform_dynamicContent_entry_filters_entry_filters_entry","translation_domain":null,"cache_key":"_emailform_dynamicContent_entry_filters_entry_filters_entry_dynamic_content_filter_entry_filters","errors":"[object] (Symfony\\Component\\Form\\FormErrorIterator: )","valid":true,"data":null,"required":true,"size":null,"label_attr":[],"compound":true,"method":"POST","action":"","submitted":false,"mauticContent":"email","mauticTemplate":"MauticEmailBundle:FormTheme\\Email:_emailform_dynamicContent_entry_filters_entry_filters_entry_widget.html.php","view":"[object] (Mautic\\CoreBundle\\Templating\\Engine\\PhpEngine: {})"},"app":"[object] (Symfony\\Bundle\\FrameworkBundle\\Templating\\GlobalVariables: {})","value":null,"attr":{"class":"form-control"},"form":"[object] (Symfony\\Component\\Form\\FormView: null)","id":"emailform_dynamicContent_0_filters_0_filters___name__","name":"__name__","full_name":"emailform[dynamicContent][0][filters][0][filters][__name__]","disabled":false,"label":false,"label_format":null,"multipart":false,"block_prefixes":["form","dynamic_content_filter_entry_filters","_emailform_dynamicContent_entry_filters_entry_filters_entry"],"unique_block_prefix":"_emailform_dynamicContent_entry_filters_entry_filters_entry","translation_domain":null,"cache_key":"_emailform_dynamicContent_entry_filters_entry_filters_entry_dynamic_content_filter_entry_filters","errors":"[object] (Symfony\\Component\\Form\\FormErrorIterator: )","valid":true,"data":null,"required":true,"size":null,"label_attr":[],"compound":true,"method":"POST","action":"","submitted":false,"mauticContent":"email","mauticTemplate":"MauticEmailBundle:FormTheme\\Email:_emailform_dynamicContent_entry_filters_entry_filters_entry_widget.html.php","view":"[object] (Mautic\\CoreBundle\\Templating\\Engine\\PhpEngine: {})","isPrototype":true,"filterType":"","filterGroup":""} []
`

**Comment by RCheesley** [comment]
I also have the same issue when trying to add a new report, here is the stack trace:

```
[2021-03-13 19:47:49] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ContextErrorException: "PHP Notice - Trying to access array offset on value of type null" at /var/www/html/app/bundles/ReportBundle/Form/Type/FilterSelectorType.php line 80 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ContextErrorException(code: 0): PHP Notice - Trying to access array offset on value of type null at /var/www/html/app/bundles/ReportBundle/Form/Type/FilterSelectorType.php:80)
[stacktrace]
#0 /var/www/html/app/bundles/ReportBundle/Form/Type/FilterSelectorType.php(80): Mautic\\CoreBundle\\ErrorHandler\\ErrorHandler->handleError(8, 'PHP Notice - Tr...', '/var/www/html/a...', 80, Array)
#1 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(214): Mautic\\ReportBundle\\Form\\Type\\FilterSelectorType->Mautic\\ReportBundle\\Form\\Type\\{closure}(Object(Symfony\\Component\\Form\\FormEvent), 'form.pre_set_da...', Object(Symfony\\Component\\EventDispatcher\\EventDispatcher))
#2 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\\Component\\EventDispatcher\\EventDispatcher->doDispatch(Array, 'form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#3 /var/www/html/vendor/symfony/event-dispatcher/ImmutableEventDispatcher.php(33): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#4 /var/www/html/vendor/symfony/form/Form.php(336): Symfony\\Component\\EventDispatcher\\ImmutableEventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#5 /var/www/html/vendor/symfony/form/Form.php(476): Symfony\\Component\\Form\\Form->setData(NULL)
#6 /var/www/html/vendor/symfony/form/FormBuilder.php(210): Symfony\\Component\\Form\\Form->initialize()
#7 /var/www/html/vendor/symfony/form/Extension/Core/Type/CollectionType.php(40): Symfony\\Component\\Form\\FormBuilder->getForm()
#8 /var/www/html/vendor/symfony/form/ResolvedFormType.php(126): Symfony\\Component\\Form\\Extension\\Core\\Type\\CollectionType->buildForm(Object(Ivory\\OrderedForm\\Builder\\OrderedFormBuilder), Array)
#9 /var/www/html/vendor/symfony/form/ResolvedFormType.php(123): Symfony\\Component\\Form\\ResolvedFormType->buildForm(Object(Ivory\\OrderedForm\\Builder\\OrderedFormBuilder), Array)
#10 /var/www/html/vendor/symfony/form/FormFactory.php(80): Symfony\\Component\\Form\\ResolvedFormType->buildForm(Object(Ivory\\OrderedForm\\Builder\\OrderedFormBuilder), Array)
#11 /var/www/html/vendor/symfony/form/FormFactory.php(38): Symfony\\Component\\Form\\FormFactory->createNamedBuilder('filters', Object(Ivory\\OrderedForm\\OrderedResolvedFormType), NULL, Array)
#12 /var/www/html/vendor/symfony/form/Form.php(839): Symfony\\Component\\Form\\FormFactory->createNamed('filters', 'Mautic\\\\ReportBu...', NULL, Array)
#13 /var/www/html/app/bundles/ReportBundle/Form/Type/ReportType.php(222): Symfony\\Component\\Form\\Form->add('filters', 'Mautic\\\\ReportBu...', Array)
#14 /var/www/html/app/bundles/ReportBundle/Form/Type/ReportType.php(395): Mautic\\ReportBundle\\Form\\Type\\ReportType->Mautic\\ReportBundle\\Form\\Type\\{closure}(Object(Symfony\\Component\\Form\\Form), 'assets', Array, Array, Object(Mautic\\ReportBundle\\Entity\\Report))
#15 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(214): Mautic\\ReportBundle\\Form\\Type\\ReportType->Mautic\\ReportBundle\\Form\\Type\\{closure}(Object(Symfony\\Component\\Form\\FormEvent), 'form.pre_set_da...', Object(Symfony\\Component\\EventDispatcher\\EventDispatcher))
#16 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\\Component\\EventDispatcher\\EventDispatcher->doDispatch(Array, 'form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#17 /var/www/html/vendor/symfony/event-dispatcher/ImmutableEventDispatcher.php(33): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#18 /var/www/html/vendor/symfony/form/Form.php(336): Symfony\\Component\\EventDispatcher\\ImmutableEventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#19 /var/www/html/vendor/symfony/form/Form.php(476): Symfony\\Component\\Form\\Form->setData(Object(Mautic\\ReportBundle\\Entity\\Report))
#20 /var/www/html/vendor/symfony/form/FormBuilder.php(210): Symfony\\Component\\Form\\Form->initialize()
#21 /var/www/html/app/bundles/ReportBundle/Generator/ReportGenerator.php(103): Symfony\\Component\\Form\\FormBuilder->getForm()
#22 /var/www/html/app/bundles/ReportBundle/Model/ReportModel.php(162): Mautic\\ReportBundle\\Generator\\ReportGenerator->getForm(Object(Mautic\\ReportBundle\\Entity\\Report), Array)
#23 /var/www/html/app/bundles/ReportBundle/Controller/ReportController.php(467): Mautic\\ReportBundle\\Model\\ReportModel->createForm(Object(Mautic\\ReportBundle\\Entity\\Report), Object(Symfony\\Component\\Form\\FormFactory), '/index_dev.php/...')
#24 /var/www/html/app/bundles/CoreBundle/Controller/CommonController.php(456): Mautic\\ReportBundle\\Controller\\ReportController->newAction(Object(Mautic\\ReportBundle\\Entity\\Report), '')
#25 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(151): Mautic\\CoreBundle\\Controller\\CommonController->executeAction('new', 0, 0, '')
#26 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(68): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)
#27 /var/www/html/vendor/symfony/http-kernel/Kernel.php(200): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#28 /var/www/html/app/AppKernel.php(104): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#29 /var/www/html/app/middlewares/CORSMiddleware.php(91): AppKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#30 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(43): Mautic\\Middleware\\CORSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#31 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(61): Mautic\\Middleware\\CatchExceptionMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#32 /var/www/html/app/middlewares/VersionCheckMiddleware.php(67): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#33 /var/www/html/app/middlewares/TrustMiddleware.php(51): Mautic\\Middleware\\VersionCheckMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#34 /var/www/html/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Mautic\\Middleware\\TrustMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#35 /var/www/html/vendor/stack/run/src/Stack/run.php(13): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))
#36 /var/www/html/index_dev.php(25): Stack\
un(Object(Stack\\StackedHttpKernel))
#37 {main}
"} 

```

**Comment by RCheesley** [comment]
Same error when saving a segment that had a filter of 'is not empty' on a custom date/time field:

```
[2021-03-13 21:18:10] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ContextErrorException: "PHP Notice - Trying to access array offset on value of type null" at /var/www/html/app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php line 87 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ContextErrorException(code: 0): PHP Notice - Trying to access array offset on value of type null at /var/www/html/app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php:87)
[stacktrace]
#0 /var/www/html/app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php(87): Mautic\\CoreBundle\\ErrorHandler\\ErrorHandler->handleError(8, 'PHP Notice - Tr...', '/var/www/html/a...', 87, Array)
#1 /var/www/html/vendor/symfony/form/Form.php(1055): Mautic\\LeadBundle\\Form\\DataTransformer\\FieldFilterTransformer->reverseTransform(Array)
#2 /var/www/html/vendor/symfony/form/Form.php(643): Symfony\\Component\\Form\\Form->normToModel(Array)
#3 /var/www/html/vendor/symfony/form/Form.php(573): Symfony\\Component\\Form\\Form->submit(Array, true)
#4 /var/www/html/vendor/symfony/form/Extension/HttpFoundation/HttpFoundationRequestHandler.php(109): Symfony\\Component\\Form\\Form->submit(Array, true)
#5 /var/www/html/vendor/symfony/form/Form.php(487): Symfony\\Component\\Form\\Extension\\HttpFoundation\\HttpFoundationRequestHandler->handleRequest(Object(Symfony\\Component\\Form\\Form), Object(Symfony\\Component\\HttpFoundation\\Request))
#6 /var/www/html/app/bundles/CoreBundle/Controller/AbstractFormController.php(167): Symfony\\Component\\Form\\Form->handleRequest(Object(Symfony\\Component\\HttpFoundation\\Request))
#7 /var/www/html/app/bundles/LeadBundle/Controller/ListController.php(180): Mautic\\CoreBundle\\Controller\\AbstractFormController->isFormValid(Object(Symfony\\Component\\Form\\Form))
#8 /var/www/html/app/bundles/CoreBundle/Controller/CommonController.php(456): Mautic\\LeadBundle\\Controller\\ListController->newAction(0, '')
#9 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(151): Mautic\\CoreBundle\\Controller\\CommonController->executeAction('new', 0, 0, '')
#10 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(68): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)
#11 /var/www/html/vendor/symfony/http-kernel/Kernel.php(200): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#12 /var/www/html/app/AppKernel.php(104): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#13 /var/www/html/app/middlewares/CORSMiddleware.php(91): AppKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#14 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(43): Mautic\\Middleware\\CORSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#15 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(61): Mautic\\Middleware\\CatchExceptionMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#16 /var/www/html/app/middlewares/VersionCheckMiddleware.php(67): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#17 /var/www/html/app/middlewares/TrustMiddleware.php(51): Mautic\\Middleware\\VersionCheckMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#18 /var/www/html/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Mautic\\Middleware\\TrustMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#19 /var/www/html/vendor/stack/run/src/Stack/run.php(13): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))
#20 /var/www/html/index_dev.php(25): Stack\
un(Object(Stack\\StackedHttpKernel))
#21 {main}
"} 
```

**Comment by mohit-rocks** [comment]
@RCheesley 
After this PR change, I am not able to reproduce the other two issues on local.
I think we are good to open this PR for testing.

**Comment by RCheesley** [comment]
@mohit-rocks great - thanks for tracking this down and getting to the bottom of it!

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Community Forums**. There might be relevant details there:

https://forum.mautic.org/t/notices-trying-to-access-array-offset-on-value-of-type-null/18008/9

**Comment by RCheesley** [comment]
@mohit-rocks This seems to resolve the issue for email - creating a new email using the Goldstar theme I no longer see the errors. 

DWC does seem to have been fixed as that is no longer throwing an error for me.

When I checked with reports, if I have dev mode enabled and I edit the report then I still have this error being thrown:

```
[2021-03-19 19:08:34] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ContextErrorException: "PHP Notice - Trying to access array offset on value of type null" at /var/www/html/app/bundles/ReportBundle/Form/Type/FilterSelectorType.php line 80 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ContextErrorException(code: 0): PHP Notice - Trying to access array offset on value of type null at /var/www/html/app/bundles/ReportBundle/Form/Type/FilterSelectorType.php:80)
[stacktrace]
#0 /var/www/html/app/bundles/ReportBundle/Form/Type/FilterSelectorType.php(80): Mautic\\CoreBundle\\ErrorHandler\\ErrorHandler->handleError(8, 'PHP Notice - Tr...', '/var/www/html/a...', 80, Array)
#1 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(214): Mautic\\ReportBundle\\Form\\Type\\FilterSelectorType->Mautic\\ReportBundle\\Form\\Type\\{closure}(Object(Symfony\\Component\\Form\\FormEvent), 'form.pre_set_da...', Object(Symfony\\Component\\EventDispatcher\\EventDispatcher))
#2 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\\Component\\EventDispatcher\\EventDispatcher->doDispatch(Array, 'form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#3 /var/www/html/vendor/symfony/event-dispatcher/ImmutableEventDispatcher.php(33): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#4 /var/www/html/vendor/symfony/form/Form.php(336): Symfony\\Component\\EventDispatcher\\ImmutableEventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#5 /var/www/html/vendor/symfony/form/Form.php(476): Symfony\\Component\\Form\\Form->setData(NULL)
#6 /var/www/html/vendor/symfony/form/FormBuilder.php(210): Symfony\\Component\\Form\\Form->initialize()
#7 /var/www/html/vendor/symfony/form/Extension/Core/Type/CollectionType.php(40): Symfony\\Component\\Form\\FormBuilder->getForm()
#8 /var/www/html/vendor/symfony/form/ResolvedFormType.php(126): Symfony\\Component\\Form\\Extension\\Core\\Type\\CollectionType->buildForm(Object(Ivory\\OrderedForm\\Builder\\OrderedFormBuilder), Array)
#9 /var/www/html/vendor/symfony/form/ResolvedFormType.php(123): Symfony\\Component\\Form\\ResolvedFormType->buildForm(Object(Ivory\\OrderedForm\\Builder\\OrderedFormBuilder), Array)
#10 /var/www/html/vendor/symfony/form/FormFactory.php(80): Symfony\\Component\\Form\\ResolvedFormType->buildForm(Object(Ivory\\OrderedForm\\Builder\\OrderedFormBuilder), Array)
#11 /var/www/html/vendor/symfony/form/FormFactory.php(38): Symfony\\Component\\Form\\FormFactory->createNamedBuilder('filters', Object(Ivory\\OrderedForm\\OrderedResolvedFormType), NULL, Array)
#12 /var/www/html/vendor/symfony/form/Form.php(839): Symfony\\Component\\Form\\FormFactory->createNamed('filters', 'Mautic\\\\ReportBu...', NULL, Array)
#13 /var/www/html/app/bundles/ReportBundle/Form/Type/ReportType.php(222): Symfony\\Component\\Form\\Form->add('filters', 'Mautic\\\\ReportBu...', Array)
#14 /var/www/html/app/bundles/ReportBundle/Form/Type/ReportType.php(395): Mautic\\ReportBundle\\Form\\Type\\ReportType->Mautic\\ReportBundle\\Form\\Type\\{closure}(Object(Symfony\\Component\\Form\\Form), 'form.submission...', Array, Array, Object(Mautic\\ReportBundle\\Entity\\Report))
#15 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(214): Mautic\\ReportBundle\\Form\\Type\\ReportType->Mautic\\ReportBundle\\Form\\Type\\{closure}(Object(Symfony\\Component\\Form\\FormEvent), 'form.pre_set_da...', Object(Symfony\\Component\\EventDispatcher\\EventDispatcher))
#16 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\\Component\\EventDispatcher\\EventDispatcher->doDispatch(Array, 'form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#17 /var/www/html/vendor/symfony/event-dispatcher/ImmutableEventDispatcher.php(33): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#18 /var/www/html/vendor/symfony/form/Form.php(336): Symfony\\Component\\EventDispatcher\\ImmutableEventDispatcher->dispatch('form.pre_set_da...', Object(Symfony\\Component\\Form\\FormEvent))
#19 /var/www/html/vendor/symfony/form/Form.php(476): Symfony\\Component\\Form\\Form->setData(Object(Mautic\\ReportBundle\\Entity\\Report))
#20 /var/www/html/vendor/symfony/form/FormBuilder.php(210): Symfony\\Component\\Form\\Form->initialize()
#21 /var/www/html/app/bundles/ReportBundle/Generator/ReportGenerator.php(103): Symfony\\Component\\Form\\FormBuilder->getForm()
#22 /var/www/html/app/bundles/ReportBundle/Model/ReportModel.php(162): Mautic\\ReportBundle\\Generator\\ReportGenerator->getForm(Object(Mautic\\ReportBundle\\Entity\\Report), Array)
#23 /var/www/html/app/bundles/ReportBundle/Controller/ReportController.php(343): Mautic\\ReportBundle\\Model\\ReportModel->createForm(Object(Mautic\\ReportBundle\\Entity\\Report), Object(Symfony\\Component\\Form\\FormFactory), '/index_dev.php/...')
#24 /var/www/html/app/bundles/CoreBundle/Controller/CommonController.php(456): Mautic\\ReportBundle\\Controller\\ReportController->editAction('23', '')
#25 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(151): Mautic\\CoreBundle\\Controller\\CommonController->executeAction('edit', '23', 0, '')
#26 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(68): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)
#27 /var/www/html/vendor/symfony/http-kernel/Kernel.php(200): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#28 /var/www/html/app/AppKernel.php(104): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#29 /var/www/html/app/middlewares/CORSMiddleware.php(91): AppKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#30 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(43): Mautic\\Middleware\\CORSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#31 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(61): Mautic\\Middleware\\CatchExceptionMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#32 /var/www/html/app/middlewares/VersionCheckMiddleware.php(67): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#33 /var/www/html/app/middlewares/TrustMiddleware.php(51): Mautic\\Middleware\\VersionCheckMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#34 /var/www/html/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Mautic\\Middleware\\TrustMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#35 /var/www/html/vendor/stack/run/src/Stack/run.php(13): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))
#36 /var/www/html/index_dev.php(25): Stack\
un(Object(Stack\\StackedHttpKernel))
#37 {main}
"} 
```

**Comment by viniciusferrao** [comment]
Is there any available workaround for those on 3.3 branch? Or is it mandatory to upgrade to 4.0 even while this is still on RC?

**Comment by mohit-rocks** [comment]
@viniciusferrao Can you please try this patch on 3.3 branch https://patch-diff.githubusercontent.com/raw/mautic/mautic/pull/9795.patch
Let me know if it doesn't work, I can share another patch.

**Comment by viniciusferrao** [comment]
Unfortunately the issue persisted. I'm not sure whats happening anymore to be honest.

**Comment by RCheesley** [comment]
@viniciusferrao if this is with Mautic 4 please raise another issue, explicitly stating what is causing a 500 error and including the logs per the post template. Otherwise, these issues will be addressed in the next stable release, 4.0.0.

**Comment by viniciusferrao** [comment]
Thanks @RCheesley but it's still 3.3

Do you recommend to just upgrade to 4.0 even in beta?

**Comment by RCheesley** [comment]
It is in RC and we are just waiting on a couple of fixes before the main release is made. Once that is done, there will be no more releases on the 3.x series unless there are security fixes. So yes, once you are ready, updating will be necessary. By all means apply this patch yourself.

**Comment by kuzmany** [comment]
This should fixed it

https://github.com/mautic/mautic/pull/10806

Same issue in segment filter

**Comment by RCheesley** [comment]
Closing as these issues are now resolved.

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Community Forums**. There might be relevant details there:

https://forum.mautic.org/t/dynamic-content-in-email-sent-from-campaign-is-not-replaced/26058/1
